### PR TITLE
Injected ITypeCatalog into Diagnostics 

### PR DESCRIPTION
**Pull request by jchannon**
injected ITypeCatalog into Diagnostics and added access to nancy diagnostics todemo.hosting.self for mono users :)

address infomodule in #2221

### CODE REVIEW

File: src/Nancy/Diagnostics/DiagnosticsHook.cs [hunk 1]
```diff
@@ -125,9 +125,9 @@ private static INancyEnvironment GetDiagnosticsEnvironment()
         private static bool ValidateConfiguration(DiagnosticsConfiguration configuration)
         {
             return !string.IsNullOrWhiteSpace(configuration.Password) &&
-                !string.IsNullOrWhiteSpace(configuration.CookieName) &&
-                !string.IsNullOrWhiteSpace(configuration.Path) &&
-                configuration.SlidingTimeout != 0;
+            !string.IsNullOrWhiteSpace(configuration.CookieName) &&
```
Comment (thecodejunkie) [inline src/Nancy/Diagnostics/DiagnosticsHook.cs]: Bring back the indention here

File: src/Nancy/Diagnostics/DiagnosticsHook.cs [hunk 2]
```diff
@@ -283,8 +283,8 @@ private static DiagnosticsSession ProcessLogin(NancyContext context, Diagnostics
         private static bool IsLoginRequest(NancyContext context, DiagnosticsConfiguration diagnosticsConfiguration)
         {
             return context.Request.Method.Equals("POST", StringComparison.OrdinalIgnoreCase) &&
-                context.Request.Url.BasePath.TrimEnd('/').EndsWith(diagnosticsConfiguration.Path) &&
-                context.Request.Url.Path == "/";
+            context.Request.Url.BasePath.TrimEnd('/').EndsWith(diagnosticsConfiguration.Path) &&
```
Comment (thecodejunkie) [inline src/Nancy/Diagnostics/DiagnosticsHook.cs]: Bring back the indention here

File: src/Nancy/Diagnostics/DiagnosticsModuleCatalog.cs [hunk 3]
```diff
@@ -54,18 +54,19 @@ private static TinyIoCContainer ConfigureContainer(IEnumerable<IDiagnosticsProvi
...
             diagContainer.Register<ISerializer>(new DefaultJsonSerializer(diagnosticsEnvironment));
+            diagContainer.Register<ITypeCatalog>(typeCatalog);
 
             foreach (var diagnosticsProvider in providers)
             {
                 var key = string.Concat(
-                    diagnosticsProvider.GetType().FullName,
-                    "_",
-                    diagnosticsProvider.DiagnosticObject.GetType().FullName);
+                              diagnosticsProvider.GetType().FullName,
```
Comment (thecodejunkie) [inline src/Nancy/Diagnostics/DiagnosticsModuleCatalog.cs]: Remove added indention

**Comment by thecodejunkie** [comment]
White space changes other than that LGTM

**Comment by jchannon** [comment]
Srly?

**Comment by thecodejunkie** [comment]
What? Since when did we not pay attention to these kinds of changes?

**Comment by thecodejunkie** [comment]
All of my comments are also on multi-line changes but I only added the comment on the first line

**Comment by jchannon** [comment]
Updated

**Comment by khellang** [comment]
Please revert this. Now master is borked :disappointed:

**Pull request merged**